source/AI.cpp

```diff
@@ -783,19 +783,20 @@ void AI::Step(const PlayerInfo &player, Command &activeCommands)
 			}
 			// Fighters and drones should assist their parent's mining operation if they cannot
 			// carry ore, and the asteroid is near enough that the parent can harvest the ore.
-			const shared_ptr<Minable> &minable = parent ? parent->GetTargetAsteroid() : nullptr;
-			if(it->CanBeCarried() && parent && miningTime[&*parent] < 3601 && minable
-					&& minable->Position().Distance(parent->Position()) < 600.)
+			if(it->CanBeCarried() && parent && miningTime[&*parent] < 3601)
```

Comment:
```suggestion
			if(it->CanBeCarried() && parent && miningTime[parent.get()] < 3601)
```

while we're at it, saves an unnecessary dereference.

Reply:
Done, thanks.